pkg/helper/helpers.go

```diff
@@ -198,6 +208,34 @@ func DeleteAppliedResources(resources []workapiv1.AppliedManifestResourceMeta, r
 			continue
 		}
 
+		existingOwner := u.GetOwnerReferences()
+
+		// If it is not owned by us, skip
+		if !IsOwnedBy(owner, existingOwner) {
+			continue
+		}
+
+		// Merge with the existing owners to move the owner.
+		modified := resourcemerge.BoolPtr(false)
+		resourcemerge.MergeOwnerRefs(modified, &existingOwner, []metav1.OwnerReference{*ownerCopy})
+
+		// If there are still existing owners, update ownerrefs only.
+		if len(existingOwner) > 0 {
```

Comment:
Do you want the behavior to be "any other owner" or to be "any other manifestWork  owner"?

If you could update the comment to indicate an explicit choice.

Reply:
I think it is any other owner, not only manifestWork. I will update the comment.